Improve HTML and CSS perf docs (#27558)
mdn/content: files/en-us/glossary/render_blocking/index.md

---
title: Render-blocking
slug: Glossary/Render_blocking
page-type: glossary-definition
---

{{GlossarySidebar}}

> **Render-blocking** refers to any part of the process of loading a website when it blocks the rendering of the user interface. Render-blocking is bad for web performance because it increases the length of time until users can interact with the site. This can mean being able to view content, or interact with controls.

Reviewer: ```suggestion
**Render-blocking** refers to any part of the process of loading a website when it blocks the rendering of the user interface. Render-blocking is bad for web performance because it increases the length of time until users can interact with the site. This can mean increased time before users can view the site content or interact with controls.
```

Author: "increases the length of time" ... "increased time" felt a bit repetitious. I've handled this a different way.

The most common causes of render-blocking are initially-loaded CSS or JavaScript files.

## See also

- [CSS performance optimization](/en-US/docs/Learn/Performance/CSS)
- [JavaScript performance optimization](/en-US/docs/Learn/Performance/JavaScript)

> - [What Are Render-Blocking Resources & How to Fix Render Blocking Issues](https://speckyboy.com/render-blocking/), Speckyboy, 2018

Reviewer: ```suggestion
- [What Are Render-Blocking Resources & How to Fix Render Blocking Issues](https://speckyboy.com/render-blocking/) on Speckyboy (2018)
```
